task_processing/runners/async.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from threading import Thread
+
+from task_processing.runners.runner import Runner
+
+
+class AsyncError(Exception):
+    pass
+
+
+class Async(Runner):
+    def __init__(self, executor, callbacks=None):
+        if not callbacks:
+            raise AsyncError("must provide at least one callback")
+
+        self.callbacks = callbacks
+        self.executor = executor
+
+        callback_t = Thread(target=self._callback_loop)
+        callback_t.daemon = True
+        callback_t.start()
+
+    def run(self, task_config):
+        return self.executor.run(task_config)
+
+    def _callback_loop(self):
+        event_queue = self.executor.get_event_queue()
+        while True:
+            event = event_queue.get()
```

Comment:
Since self.executor.get_event_queue() is gonna be shared between multiple objects Async/Sync/Subscription/Promise, the task update that we get in this queue may not be for this object. We need to check if the task update is for this object or not. If not, we must re-enqueue this update to self.executor.get_event_queue(). We also need locking to make sure that only one Async object is getting the lock. Maybe we don't need locking because of GIL? ¯\_(ツ)_/¯